.github/workflows/post-merge.yml

```diff
@@ -323,6 +340,52 @@ jobs:
           echo "Building the code"
           make build
 
+      - name: Run extraction
+        if: ${{ inputs.run_extraction}}
+        working-directory: ${{ inputs.project_folder }}
+        shell: bash
+        run: |
+         echo "extract packages"
+         make extraction
+
+      - name: Upload Assets
+        if: ${{ inputs.run_upload_assets}}
+        working-directory: ${{ inputs.project_folder }}
+        env:
+          EXTRACTION_DIR: "${{ inputs.extraction_dir }}"
+        run: |
+          sudo apt update
+          sudo apt install -y software-properties-common
+          curl -fsSL https://cli.github.com/packages/githubcli-archive-keyring.gpg | sudo tee /usr/share/keyrings/githubcli-archive-keyring.gpg > /dev/null
+          echo "deb [arch=$(dpkg --print-architecture) signed-by=/usr/share/keyrings/githubcli-archive-keyring.gpg] https://cli.github.com/packages stable main" | sudo tee /etc/apt/sources.list.d/github-cli.list > /dev/null
+          sudo apt update
+          sudo apt install gh
+
+          # Set environment variables
+          export GITHUB_TOKEN=${{ secrets.SYS_ORCH_GITHUB }}
+          cd ${{ inputs.project_folder }}
+
+          echo "Using GitHub token: $GITHUB_TOKEN"
+
+          # Fetch the latest tag using the project folder
+          latest_tag=$(git tag --list "${{ inputs.project_folder }}/*" | sort -V | tail -n 1)
+          echo "Latest tag: $latest_tag"
+          echo "Attempting to create a release for tag: $latest_tag"
+
+          # Create a release for the latest tag
+          gh release create "$latest_tag" --title "Release $latest_tag" --notes "Release notes for $latest_tag"
+          echo "Release creation command executed for tag: $latest_tag"
+
+          for file in "$EXTRACTION_DIR"/*; do
+            if [ -f "$file" ]; then
+              echo "Uploading $file to release assets..."
+              gh release upload "$latest_tag" "$file" --clobber
+              echo "$file uploaded to release assets."
+            else
+              echo "$file not found, skipping upload."
+            fi
+          done
```

Comment:
Maybe it makes sense to check at the very beginning of this step whether the `EXTRACTION_DIR` directory has been provided and exists. And if not, to fail the step immediately?